src/screens/Form/index.tsx

```diff
@@ -1,57 +1,92 @@
+import { Alert, KeyboardAvoidingView, Platform, ScrollView, Text } from 'react-native';
 import { GestureHandlerRootView, TouchableNativeFeedback } from 'react-native-gesture-handler';
 import { theme } from 'utils/theme';
 import { DatePicker } from '../../components/DatePickerApp';
 import { FormHeader } from '../../components/FormHeader';
 import { Input } from '../../components/Input';
 import { useNavigation } from '@react-navigation/native';
+import { makeViagemEntradaService, makeViagemService } from 'core';
+import { ViagemModel } from 'core/database/models';
 
-const data = [
-  {
-    id: 1,
-    local: 'Pitangueiras',
-    data: new Date('2022-01-01'),
-    descricao:
-      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione aut commodi optio. Nostrum, numquam magni.',
-  },
-  {
-    id: 2,
-    local: 'Pitangueiras',
-    data: new Date('2022-01-01'),
-    descricao:
-      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione aut commodi optio. Nostrum, numquam magni.',
-  },
-  {
-    id: 3,
-    local: 'Pitangueiras',
-    data: new Date('2022-01-01'),
-    descricao:
-      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione aut commodi optio. Nostrum, numquam magni.',
-  },
-  {
-    id: 4,
-    local: 'Pitangueiras',
-    data: new Date('2022-01-01'),
-    descricao:
-      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ratione aut commodi optio. Nostrum, numquam magni.',
-  },
-];
+const viagemService = makeViagemService();
+const viagemEntradaService = makeViagemEntradaService();
 
 export function Form({ route }) {
   const navigation = useNavigation();
   const { params } = route;
   const { apenasConsulta, title } = params;
-
+  const [dateValue, setDateValue] = useState<Date>();
+  const [entradas, setEntradas] = useState([]);
+  const [viagem, setViagem] = useState<ViagemModel>(params);
   const [local, setLocal] = useState('');
 
   const handleDateChange = (date: Date) => {
-    // A função de retorno de chamada para atualizar o valor do campo de data
+    setDateValue(new Date(date));
   };
 
-  function handleSalvarViagem() {
-    console.log('salvar');
+  useEffect(() => {
+    if (params.local !== null || params.local !== undefined) {
+      setLocal(params.local);
+    }
+    // Verifica se estamos em modo de consulta e obtém as entradas se necessário
+    if (apenasConsulta) {
+      void obterEntradas();
+    }
+  }, [apenasConsulta]);
+
+  async function obterEntradas() {
+    // Chama o serviço para obter as entradas
+    const entradasViagem = await viagemEntradaService.obterTodas();
+    setEntradas(entradasViagem);
+  }
+  async function handleSalvarViagem() {
+    if (!local) {
+      alert('Por favor, informe o local da viagem.');
+      return;
+    }
+
+    if (!dateValue) {
+      alert('Por favor, selecione a data da viagem.');
+      return;
+    }
+
+    if (viagem.id === null || viagem.id === undefined) {
```

Comment:
Dica: podemos usar o ! para validar se não tem valor
Isso evitara a necessidade de comparar com nulo ou indefinido
```suggestion
    if (!viagem.id) {
```